patroni/dcs/kubernetes.py

```diff
@@ -321,7 +321,7 @@ def set_failover_value(self, value, index=None):
         """Unused"""
 
     def manual_failover(self, leader, candidate, scheduled_at=None, index=None):
-        annotations = {'leader': leader or None, 'member': candidate or None, 'scheduled_at': scheduled_at}
+        annotations = {'leader': leader or '', 'member': candidate or '', 'scheduled_at': scheduled_at or ''}
```

Comment:
Kubernetes doesn't like annotations with null values when you create a new object (ConfigMap or Endpoint), therefore we are removing them here: https://github.com/zalando/patroni/pull/782/files#diff-87299faee0431ddce0848d6ff7726df7R264

But in case of PATCH request, null values have a different semantic. If a given annotation has null value in the PATCH request, Kubernetes supposed to remove such annotation from the object.

You are replacing null values with empty string, and as a result annotations are not removed:
```yaml
metadata:
  annotations:
    leader: ""
    member: ""
    scheduled_at: ""
```

From one side Patroni is able to handle it correctly (and acceptance tests prove that), but from another side, this is not how it supposed to work.

Unfortunately I don't have access to openshift clusters and can't investigate further.

Reply:
Ok, I got Your point, will investigate a little bit more.

Reply:
well, that is issue with kubernetes. just tested on oshit 3.9 (kubernetes 1.9) everything works fine, closing this PR.

Reply:
That's very interesting.
I think I tested it on all versions starting from at least kubernetes 1.7 and it was working fine.
Which version does it fail on?

Reply:
server versions
```
openshift v1.5.1
kubernetes v1.5.2+43a9be4
```